session/autocharthandler.go

```diff
@@ -40,8 +40,12 @@ func (instance *AutoChartInstance) SetObjectAndEvents(sessionState *State, actio
 }
 
 func (instance *AutoChartInstance) GetObjectDefinition(objectType string) (string, senseobjdef.SelectType, senseobjdef.DataDefType, error) {
-	// Todo error checking
-	return instance.ObjectDef.Select.Path, instance.ObjectDef.Select.Type, instance.ObjectDef.DataDef.Type, nil
+	// Checking if selectable
+	if instance.ObjectDef.Select != nil {
+		return instance.ObjectDef.Select.Path, instance.ObjectDef.Select.Type, instance.ObjectDef.DataDef.Type, nil
+	}
+	//Nothing selectable
+	return "", senseobjdef.SelectTypeUnknown, senseobjdef.DataDefUnknown, errors.New("Autochart object has no selectable option")
```

Comment:
Is datadef always nil when select is?

I'd also flip the table and return the error in the if statement

Reply:
no datadef should always be set. and as per previous comment, think this is the wrong place to do the check, changesheet etc would give an error getting the data because there's no "select"  I think?